### PR TITLE
Workaround RecursionError on profile data

### PR DESCRIPTION
**Pull request by TomAugspurger**
@mrocklin would be curious to hear of any alternatives. This is very much a workaround rather than a proper fix.

### CODE REVIEW

File: distributed/tests/test_profile.py [hunk 1]
```diff
@@ -199,3 +202,16 @@ def stop():
...
+    # https://github.com/dask/distributed/issues/1674
+    n = 500
+    original = outer = {}
+    inner = {}
+
+    for i in range(n):
+        outer["children"] = inner
+        outer, inner = inner, {}
+    profile = Profile(original)
+    sizeof(profile)
```
Comment (TomAugspurger) [inline distributed/tests/test_profile.py]: This raises a RecursionError on `original`.

**Comment by mrocklin** [comment]
Ah, I was walking through my e-mail chronologically.  I think that this is maybe a more robust solution to the problem?  https://github.com/dask/distributed/issues/1674#issuecomment-583698574

File: distributed/comm/utils.py [hunk 2]
```diff
@@ -37,7 +38,12 @@ def _to_frames():
             logger.exception(e)
             raise
 
-    if FRAME_OFFLOAD_THRESHOLD and sizeof(msg) > FRAME_OFFLOAD_THRESHOLD:
+    try:
+        msg_size = sizeof(msg)
+    except RecursionError:
+        msg_size = sys.getsizeof(msg)
```
Comment (mrocklin) [inline distributed/comm/utils.py]: I suggest instead that we default to offloading the serialization work in this case.  This object has proven that it's odd, so it probably makes sense to move it off to another thread for safety.

```suggestion
        msg_size = math.inf
```

**Comment by mrocklin** [comment]
Thanks @TomAugspurger !

**Pull request merged**